frontend/app/onboard.tsx

```diff
@@ -0,0 +1,245 @@
+    question: "What is your email?",
+    placeholder: "Enter Email",
+  },
+  {
+    id: 2,
+    title: "Account Information",
+    question: "Pick a password",
+    placeholder: "Enter Password",
+  },
+  {
+    id: 3,
+    title: "Account Information",
+    question: "Pick a username",
+    placeholder: "Enter Username",
+  },
+  {
+    id: 4,
+    title: "Link Music Account",
+    question: "Link to your music account",
+  },
+];
+
+const OnboardingCarousel: React.FC = () => {
+  const [currentSlide, setCurrentSlide] = useState(0); // Start from slide 0
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [username, setUsername] = useState("");
+  const [passwordMatch, setPasswordMatch] = useState(true);
+  const [inputValid, setInputValid] = useState(false);
+  const [tried, setTried] = useState(false);
+  const [passwordVisible, setPasswordVisible] = useState(false);
+
+  const progressBar1 = useSharedValue(0);
+  const progressBar2 = useSharedValue(0);
+
+  const emailFormat = /^[ ]*([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})[ ]*$/i;
+
+  useEffect(() => {
+    if (currentSlide === 2) {
+      setPasswordMatch(password === confirmPassword);
+    }
+  }, [password, confirmPassword, currentSlide]);
+
+  const handleSlideChange = (newSlideIndex: number) => {
+    setPasswordVisible(false);
+    setCurrentSlide(newSlideIndex);
+
+    progressBar1.value = withTiming(newSlideIndex > 0 ? 1 : 0, {
+      duration: 500,
+    });
+    progressBar2.value = withTiming(
+      newSlideIndex > 1 ? (newSlideIndex - 1) / 3 : 0,
+      { duration: 500 },
+    );
+  };
+
+  const handleNext = () => {
+    if (currentSlide < slides.length - 1) {
+      if (inputValid || (currentSlide === 2 && passwordMatch)) {
+        handleSlideChange(currentSlide + 1);
+        setInputValid(false);
+      } else {
+        setTried(true);
+      }
+    } else {
+      alert(
+        `Onboarding Completed! Name: ${name}, Email: ${email}, Password: ${password}, Username: ${username}`,
+      );
+      router.push("/");
+    }
+  };
+
+  return (
+    <View style={styles.container}>
+      <Header
+        title={slides[currentSlide].title}
+        subtitle={slides[currentSlide].question}
+      />
+
+      {currentSlide === 2 ? (
+        <View>
+          <TextInput
+            style={styles.input}
+            placeholder="Enter Password"
+            secureTextEntry={!passwordVisible}
+            value={password}
+            onChangeText={setPassword}
+          />
+          <TouchableOpacity
+            style={styles.eyeIcon}
+            onPress={() => setPasswordVisible(!passwordVisible)}
+          >
+            <Ionicons
+              name={passwordVisible ? "eye-off" : "eye"}
+              size={24}
+              color="gray"
+            />
+          </TouchableOpacity>
+          <TextInput
```

Comment:
Right now it is not checking if I have entered a password. I can just leave the field blank and continue the onboarding.